gnucash/gnucash-commands.cpp

```diff
@@ -579,14 +579,27 @@ Gnucash::run_scripting (int argc, char **argv,
         gnc_shutdown_cli (1);
     }
 
+    std::vector newArgv { argv[0] };
+    bool foundDoubleDash = false;
+
+    // all arguments after "--" will be sent to the guile/python
+    // script/REPL.
```

Comment:
Huh? From bash(1): " A -- signals the end of options and disables further option processing.  Any arguments after the  -- are treated as filenames and arguments.  An argument of - is equivalent to --."

Reply:
That's what I tried to explain. What's your 'Huh?' about exactly ? Note it says **filename** and arguments. And also note the intention of `--` is to separate arguments from options to gnucash-cli, not arguments to pass on to a script. 

If you repurpose it for separating out options to pass on, you loose the ability to solve potential issues with gnucash-cli parameters themselves. Perhaps the most expressive example is when a user has chosen `--Something.gnucash` as file name. Not something a dev would easily do, but it is a perfectly valid file name. In order to be able to use that filename you have to do
```
gnucash-cli -- --Something.gnucash
```
That leaves no room to now also suggest where the arguments to the script should start.

Reply:
"Huh" is about "You may get unexpected results by repurposing this special parameter." because @christopherlam isn't repurposing it. `python` and `guile` also take options and we want `boost::program_options` to not try to parse them, so using `--`&mdash;which `boost::program_options` supports already&mdash;is the right thing to do.

> And also note the intention of -- is to separate arguments from options to gnucash-cli, not arguments to pass on to a script.

That's a distinction without a difference: The arguments are for gnucash-cli to pass on to the scripting language program just like the `--script` value.

> That leaves no room to now also suggest where the arguments to the script should start.

I think that's obvious, they go after the filename; for the use case of creating a new book I guess we need to support the `--nofile` option so that the first positional parameter gets passed on to the user.